indexer/api/api.go

```diff
@@ -0,0 +1,167 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/go-chi/chi"
+)
+
+// Database interface for deposits
+type DepositDB interface {
+	GetDeposits(limit int, cursor string, sortDirection string) ([]Deposit, string, bool, error)
+}
+
+// Database interface for withdrawals
+type WithdrawalDB interface {
+	GetWithdrawals(limit int, cursor string, sortDirection string, sortBy string) ([]Withdrawal, string, bool, error)
+}
+
+// Deposit data structure
+type Deposit struct {
+	Guid            string        `json:"guid"`
+	Amount          string        `json:"amount"`
+	BlockNumber     int           `json:"blockNumber"`
+	BlockTimestamp  time.Time     `json:"blockTimestamp"`
+	From            string        `json:"from"`
+	To              string        `json:"to"`
+	TransactionHash string        `json:"transactionHash"`
+	L1Token         TokenListItem `json:"l1Token"`
+	L2Token         TokenListItem `json:"l2Token"`
+}
+
+// Withdrawal data structure
+type Withdrawal struct {
+	Guid            string        `json:"guid"`
+	Amount          string        `json:"amount"`
+	BlockNumber     int           `json:"blockNumber"`
+	BlockTimestamp  time.Time     `json:"blockTimestamp"`
+	From            string        `json:"from"`
+	To              string        `json:"to"`
+	TransactionHash string        `json:"transactionHash"`
+	WithdrawalState string        `json:"withdrawalState"`
+	Proof           *ProofClaim   `json:"proof"`
+	Claim           *ProofClaim   `json:"claim"`
+	L1Token         TokenListItem `json:"l1Token"`
+	L2Token         TokenListItem `json:"l2Token"`
+}
+
+// TokenListItem data structure
+type TokenListItem struct {
+	ChainId    int        `json:"chainId"`
+	Address    string     `json:"address"`
+	Name       string     `json:"name"`
+	Symbol     string     `json:"symbol"`
+	Decimals   int        `json:"decimals"`
+	LogoURI    string     `json:"logoURI"`
+	Extensions Extensions `json:"extensions"`
+}
+
+// Extensions data structure
+type Extensions struct {
+	OptimismBridgeAddress string `json:"optimismBridgeAddress"`
+	BridgeType            string `json:"bridgeType"`
+}
+
+// ProofClaim data structure
+type ProofClaim struct {
+	TransactionHash string    `json:"transactionHash"`
+	BlockTimestamp  time.Time `json:"blockTimestamp"`
+	BlockNumber     int       `json:"blockNumber"`
+}
+
+// PaginationResponse for paginated responses
+type PaginationResponse struct {
+	// TODO type this better
+	Data        interface{} `json:"data"`
+	Cursor      string      `json:"cursor"`
+	HasNextPage bool        `json:"hasNextPage"`
+}
+
+func getDepositsHandler(depositDB DepositDB) http.HandlerFunc {
```

Comment:
rather than utilizing closures, it might be a bit cleaner to have the handlers be methods on the API struct. Something like
```golang
type Api struct {
	Router       *chi.Mux
	DepositDB    DepositDB
	WithdrawalDB WithdrawalDB
}

func (a *API) DepositsHandler(return func(w http.ResponseWriter, r *http.Request) {
    ...
}

func NewApi(depositDB DepositDB, withdrawalDB WithdrawalDB) *Api {
	r := chi.NewRouter()
    api :=  &Api{
		Router:       r,
		DepositDB:    depositDB,
		WithdrawalDB: withdrawalDB,
	}

	r.Get("/api/v0/deposits", api.DepositHandler)
	...
}

	
```